lib/modules/merge.js

```diff
@@ -2,13 +2,24 @@ import { isPlainObject } from 'lodash/lang';
 import { isEmpty } from 'lodash/lang';
 
 export default function merge(state, stateDiff, options={}) {
-  const newState = { ...state };
-
   const {
     emptyDict='strict',
     array='replace',
   } = options;
 
+  // Empty state diffs get special handling. They shouldn't cause a new state
+  // object to be created in strict mode, but in replace mode we should return
+  // the new empty object.
+  if (!Object.keys(stateDiff).length) {
+    if (emptyDict === 'replace') {
+      return stateDiff;
+    }
+
+    return state;
```

Comment:
this logic isn't exactly correct. emptyDict has three values, 'skip', 'replace', and 'strict'. this should read:

```es6
  if (!Object.keys(stateDiff).length) {
    if (emptyDict === 'replace') {
      return stateDiff;
    } else if (emptyDict === 'skip') 
      return state;
    }
  }
```